--- mpeg4ip-1.0/lib/mp4av/Makefile.am.orig	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0/lib/mp4av/Makefile.am	2004-02-10 00:47:26.325614360 +0100
@@ -35,6 +35,7 @@
 	rfc3016.cpp \
 	rfc3119.cpp \
 	rfcisma.cpp
+libmp4av_la_LIBADD = ../mp4v2/libmp4v2.la
 
 EXTRA_DIST = libmp4av.dsp libmp4av_st.dsp
 
--- mpeg4ip-1.0/lib/msg_queue/Makefile.am.orig	2004-02-10 08:14:36.000000000 +0100
+++ mpeg4ip-1.0/lib/msg_queue/Makefile.am	2004-02-10 08:55:26.209395440 +0100
@@ -3,6 +3,7 @@
 libmsg_queue_la_SOURCES = \
 	msg_queue.cpp \
 	msg_queue.h
+libmsg_queue_la_LIBADD = @SDL_LIBS@
 
 EXTRA_DIST = \
 	libmsg_queue60.dsp
--- mpeg4ip-1.1/configure.in.orig	2004-05-17 23:55:17.000000000 +0200
+++ mpeg4ip-1.1/configure.in	2004-10-22 00:08:04.755591440 +0200
@@ -115,7 +115,7 @@
 saved_LIBS="$LIBS"
 LIBS="$LIBS -lm"
 AC_CHECK_LIB(vorbis, vorbis_bitrate_init, [LIBVORBIS_LIB="-lvorbis -lvorbisenc"])
-AC_CHECK_LIB(faad, faacDecOpen, [LIBVORBIS_LIB="$LIBVORBIS_LIB -lfaad"])
+dnl AC_CHECK_LIB(faad, faacDecOpen, [LIBVORBIS_LIB="$LIBVORBIS_LIB -lfaad"])  -- not used now
 have_a52=true
 AC_CHECK_LIB(a52, 
              a52_syncinfo, 
--- mpeg4ip-1.1/lib/SDLAudio/src/Makefile.am.orig	2004-05-12 23:15:24.000000000 +0200
+++ mpeg4ip-1.1/lib/SDLAudio/src/Makefile.am	2004-10-23 18:06:27.610465888 +0200
@@ -23,7 +23,7 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libmpeg4ipSDL_la_LIBADD =		\
 	@SDL_EXTRALIBS@		\
-	@SYSTEM_LIBS@
+	@SYSTEM_LIBS@ -lSDL
 libmpeg4ipSDL_la_DEPENDENCIES =	\
 	@SDL_EXTRALIBS@
 
--- mpeg4ip-1.1/player/plugin/rtp/isma_audio/Makefile.am.orig	2004-05-12 23:15:24.000000000 +0200
+++ mpeg4ip-1.1/player/plugin/rtp/isma_audio/Makefile.am	2004-10-23 18:10:59.730097416 +0200
@@ -15,13 +15,13 @@
 isma_rtp_plugin_la_LIBADD = \
 	$(top_srcdir)/lib/mp4av/libmp4av.la \
 	$(top_srcdir)/player/lib/mp4util/libmp4util.la \
-	-lm 
+	-lSDL -lm 
 
 isma_enc_rtp_plugin_la_LIBADD = \
         $(top_srcdir)/lib/mp4av/libmp4av.la \
         $(top_srcdir)/player/lib/mp4util/libmp4util.la \
         @ISMACRYP_LIB@ \
-        -lm
+	-lSDL -lm
 
 INCLUDES = -I$(top_srcdir)/player/src \
 	   -I$(top_srcdir)/player/lib \
--- mpeg4ip-1.1/player/src/codec/wav/Makefile.am.orig	2004-02-24 19:46:48.000000000 +0100
+++ mpeg4ip-1.1/player/src/codec/wav/Makefile.am	2004-10-23 18:21:11.522090896 +0200
@@ -5,6 +5,7 @@
 	ourwav.h \
 	wav_file.cpp \
 	wav_file.h
+wav_plugin_la_LIBADD = -lSDL
 
 libdir = @PLAYER_PLUGIN_DIR@
 
--- mpeg4ip-1.1/player/plugin/rtp/isma_video/Makefile.am.orig	2004-05-12 23:15:24.000000000 +0200
+++ mpeg4ip-1.1/player/plugin/rtp/isma_video/Makefile.am	2004-10-23 18:31:06.410654088 +0200
@@ -7,6 +7,7 @@
 
 isma_enc_video_rtp_plugin_la_LIBADD = \
         @ISMACRYP_LIB@ \
+	$(top_builddir)/player/lib/mp4util/libmp4util.la \
 	-lm 
 
 INCLUDES = -I$(top_srcdir)/player/src \
--- mpeg4ip-1.1/player/plugin/video/mpeg3/Makefile.am.orig	2004-04-01 06:32:13.000000000 +0200
+++ mpeg4ip-1.1/player/plugin/video/mpeg3/Makefile.am	2004-10-23 18:31:47.116465864 +0200
@@ -8,6 +8,7 @@
 	-lm \
 	$(top_builddir)/common/video/libmpeg32/video/libmpeg3video.la \
 	$(top_builddir)/common/video/libmpeg32/libmpeg3util.la \
+	$(top_builddir)/lib/mp4av/libmp4av.la \
 	@SDL_LIB_LIBS@
 
 INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/common/video/libmpeg32 -I$(top_srcdir)/common/video/libmpeg32/video -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp
--- mpeg4ip-1.1/player/src/codec/mp3/Makefile.am.orig	2004-02-24 22:58:38.000000000 +0100
+++ mpeg4ip-1.1/player/src/codec/mp3/Makefile.am	2004-10-23 18:32:36.034029264 +0200
@@ -16,7 +16,8 @@
 	-lm $(ID3_LIBS) \
 	$(top_builddir)/player/lib/audio/mp3/libourmp3.la \
 	$(top_builddir)/player/lib/audio/mp3util/libmp3utils.la \
-	$(top_builddir)/lib/fposrec/libfposrec.la
+	$(top_builddir)/lib/fposrec/libfposrec.la \
+	$(top_builddir)/lib/mp4av/libmp4av.la
 
 
 noinst_LTLIBRARIES = libmp3.la
