--- mpeg4ip-1.5.0.1/player/src/Makefile.am.orig	2006-03-10 19:49:44.000000000 +0100
+++ mpeg4ip-1.5.0.1/player/src/Makefile.am	2006-07-21 21:23:47.092512250 +0200
@@ -206,7 +206,7 @@
 	$(top_builddir)/lib/audio/libaudio.la \
 	$(top_builddir)/lib/ismacryp/libismacryp.la \
 	$(top_builddir)/lib/srtp/libsrtpif.la \
-	@SDL_LIBS@ @SRTPLIB@
+	@SDL_LIBS@ -lX11 @SRTPLIB@
 
 gmp4player_LDFLAGS= $(SDL_AUDIO_FLAGS)
 gmp4player_LDADD = $(mp4player_LDADD) \
