--- mpeg4ip-1.2/configure.in.orig	2004-12-04 19:41:31.933000000 +0100
+++ mpeg4ip-1.2/configure.in	2004-12-04 19:42:26.067770872 +0100
@@ -432,13 +432,6 @@
       haveglib20=no
    fi
   fi
-  if test x$haveglib20 = xno; then 
-   AM_PATH_GLIB(, [glib_gtk=yes], [glib_gtk=no])
-   if test x$glib_gtk = xyes; then
-      AM_PATH_GTK(, [glib_gtk=yes], [glib_gtk=no])
-      AC_DEFINE(HAVE_GTK, 1, [Have GTK on system])
-   fi
-  fi
 
   dnl If we're going to check for GLIB, we most likely need the real path
   dnl we need to save off the CFLAGS and CPPFLAGS for the AC_CHECK_HEADER
