--- mpeg4ip-1.0/lib/sdp/sdp_encode.c.orig	2003-12-04 22:51:05.000000000 +0000
+++ mpeg4ip-1.0/lib/sdp/sdp_encode.c	2004-02-11 09:57:43.000000000 +0000
@@ -479,8 +479,8 @@
 				sptr->orig_username == NULL ?
 				"-" : sptr->orig_username);
   sprintf(buffer, " %lld %lld IN ",
-	  sptr->session_id,
-	  sptr->session_version);
+	  (long long)(sptr->session_id),
+	  (long long)(sptr->session_version));
   ADD_STR_TO_ENCODE_WITH_RETURN(se, buffer);
   ADD_STR_TO_ENCODE_WITH_RETURN(se, sptr->create_addr_type);
   ADD_STR_TO_ENCODE_WITH_RETURN(se, " ");
--- mpeg4ip-1.0/player/src/audio_sdl.cpp.orig	2004-02-11 10:00:21.000000000 +0000
+++ mpeg4ip-1.0/player/src/audio_sdl.cpp	2004-02-11 11:07:45.000000000 +0000
@@ -354,7 +354,7 @@
 			m_last_fill_timestamp);
 	  m_last_fill_timestamp += m_msec_per_frame + 1; // fill plus extra
 	  ts_diff = ts - m_last_fill_timestamp;
-	  audio_message(LOG_DEBUG, "diff is %lld", ts_diff);
+	  audio_message(LOG_DEBUG, "diff is %lld", (long long)ts_diff);
 	} while (ts_diff > 0);
 	locked = 0;
 	if (m_audio_initialized != 0) {
--- mpeg4ip-1.0/lib/mp4v2/isma.cpp.orig	Tue May 27 23:30:45 2003
+++ mpeg4ip-1.0/lib/mp4v2/isma.cpp	Tue Feb 24 11:19:22 2004
@@ -485,7 +485,7 @@
     delete pVideoEsdProperty;
 
 	VERBOSE_ISMA(GetVerbosity(),
-		printf("OD data = %llu bytes\n", numBytes); MP4HexDump(pBytes, numBytes));
+		printf("OD data = %llu bytes\n", (long long)numBytes); MP4HexDump(pBytes, numBytes));
 
 	char* odCmdBase64 = MP4ToBase64(pBytes, numBytes);
 
@@ -748,7 +748,7 @@
 	CreateIsmaODUpdateCommandForStream(
 		pAudioEsd, pVideoEsd, ppBytes, pNumBytes);
 	VERBOSE_ISMA(GetVerbosity(),
-		printf("After CreateImsaODUpdateCommandForStream len %llu =\n", *pNumBytes); MP4HexDump(*ppBytes, *pNumBytes));
+		printf("After CreateImsaODUpdateCommandForStream len %llu =\n", (long long)*pNumBytes); MP4HexDump(*ppBytes, *pNumBytes));
 	// return SL config values to 2 (file)
 	// return ESID values to 0
 	if (pAudioSLConfigPredef) {
--- mpeg4ip-1.0/lib/mp4v2/mp4descriptor.cpp.orig	Wed Jun  5 23:50:38 2002
+++ mpeg4ip-1.0/lib/mp4v2/mp4descriptor.cpp	Tue Feb 24 13:55:02 2004
@@ -82,7 +82,7 @@
 {
 	VERBOSE_READ(pFile->GetVerbosity(),
 		printf("ReadDescriptor: pos = 0x%llx\n", 
-			pFile->GetPosition()));
+			(long long)pFile->GetPosition()));
 
 	// read tag and length
 	u_int8_t tag = pFile->ReadUInt8();
--- mpeg4ip-1.0/lib/mp4v2/mp4track.cpp.orig	Thu Dec  4 23:51:05 2003
+++ mpeg4ip-1.0/lib/mp4v2/mp4track.cpp	Tue Feb 24 15:17:38 2004
@@ -1607,8 +1607,8 @@
 			VERBOSE_EDIT(m_pFile->GetVerbosity(),
 				printf("GetSampleIdFromEditTime: when %llu "
 					"sampleId %u start %llu duration %lld\n", 
-					editWhen, sampleId, 
-					editSampleStartTime, editSampleDuration));
+					(unsigned long long)editWhen, sampleId, 
+					(unsigned long long)editSampleStartTime, (long long)editSampleDuration));
 
 			return sampleId;
 		}
--- mpeg4ip-1.0/player/plugin/video/h261/p64dump.cpp.orig	Thu Dec  4 23:49:33 2003
+++ mpeg4ip-1.0/player/plugin/video/h261/p64dump.cpp	Tue Feb 24 16:05:11 2004
@@ -159,10 +159,10 @@
 	va_start(ap, msg);
 	printf("-err: ");
 	vfprintf(stdout, msg, ap);
-	printf(" @g%d m%d %d/%d of %d/%d: %04x %04x %04x %04x|%04x\n",
+	printf(" @g%d m%d %ld/%d of %ld/%d: %04x %04x %04x %04x|%04x\n",
 		gob_, mba_,
-	       (u_char*)bs_ - (u_char*)ps_, nbb_,
-	       (u_char*)es_ - (u_char*)ps_, pebit_,
+	       (long)((u_char*)bs_ - (u_char*)ps_), nbb_,
+	       (long)((u_char*)es_ - (u_char*)ps_), pebit_,
 	       bs_[-4], bs_[-3], bs_[-2], bs_[-1], bs_[0]);
 }
 
--- mpeg4ip-1.0/player/src/media_utils.cpp.orig	Mon Oct 20 21:14:13 2003
+++ mpeg4ip-1.0/player/src/media_utils.cpp	Wed Feb 25 09:28:29 2004
@@ -544,6 +544,7 @@
 {
   char *slash, *cm;
   uint64_t prog;
+  unsigned long long progtmp;
   session_desc_t *sdp;
 
   name += strlen("iptv://");
@@ -556,10 +557,11 @@
   memcpy(cm, name, slash - name);
   cm[slash - name] = '\0';
   slash++;
-  if (sscanf(slash, "%llu", &prog) != 1) {
+  if (sscanf(slash, "%llu", &progtmp) != 1) {
     snprintf(errmsg, errlen, "Invalid iptv program");
     return -1;
   }
+  prog = progtmp;
   
   // check on-demand first
   sdp = find_sdp_for_program(cm, "iptvfiles/guide.sdf",
