--- mpeg4ip-0.9.8/lib/xvid/bitstream/x86_asm/Makefile.am.orig	2003-04-17 08:21:15.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/bitstream/x86_asm/Makefile.am	2003-04-17 08:34:01.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/lib/xvid/dct/x86_asm/Makefile.am.orig	2003-04-17 08:34:48.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/dct/x86_asm/Makefile.am	2003-04-17 08:46:53.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/lib/xvid/image/x86_asm/Makefile.am.orig	2003-04-17 08:34:48.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/image/x86_asm/Makefile.am	2003-04-17 08:47:10.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/lib/xvid/motion/x86_asm/Makefile.am.orig	2003-04-17 08:34:48.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/motion/x86_asm/Makefile.am	2003-04-17 08:47:23.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/lib/xvid/quant/x86_asm/Makefile.am.orig	2003-04-17 08:34:48.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/quant/x86_asm/Makefile.am	2003-04-17 08:47:39.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/lib/xvid/utils/x86_asm/Makefile.am.orig	2003-04-17 08:34:48.000000000 +0200
+++ mpeg4ip-0.9.8/lib/xvid/utils/x86_asm/Makefile.am	2003-04-17 08:47:53.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/common/video/libmpeg32/video/Makefile.am.orig	2003-04-17 08:49:22.000000000 +0200
+++ mpeg4ip-0.9.8/common/video/libmpeg32/video/Makefile.am	2003-04-17 18:02:14.000000000 +0200
@@ -7,7 +7,7 @@
 SUFFIXES = .asm
 
 .asm.lo:
-	$(LIBTOOL) --mode=compile $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
+	$(LIBTOOL) --mode=compile --tag=NASM $(STRIP_FPIC) $(NASM) @NASMFLAGS@ $<
 
 ###########################################################################
 
--- mpeg4ip-0.9.8/configure.in.orig	2003-04-17 18:02:46.000000000 +0200
+++ mpeg4ip-0.9.8/configure.in	2003-04-17 18:27:41.000000000 +0200
@@ -24,6 +24,7 @@
 AM_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 AC_PROG_CC
+AM_PROG_AS
 OBJC="???"
 AC_SUBST(OBJC)
 AC_PROG_CXX(c++)
