--- mpeg4ip-1.0RC3/lib/msg_queue/Makefile.am.wiget	2003-11-13 17:40:50.000000000 +0100
+++ mpeg4ip-1.0RC3/lib/msg_queue/Makefile.am	2003-12-03 17:09:50.913357229 +0100
@@ -6,5 +6,5 @@
 
 EXTRA_DIST = \
 	libmsg_queue60.dsp
-INCLUDES = -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/include
+INCLUDES = @SDL_CFLAGS@ -I$(top_srcdir)/include
 AM_CXXFLAGS = -D_REENTRANT -DNOCONTROLS -Wall -Werror
--- mpeg4ip-1.0RC3/lib/mpeg2t/Makefile.am.wiget	2003-11-13 17:40:50.000000000 +0100
+++ mpeg4ip-1.0RC3/lib/mpeg2t/Makefile.am	2003-12-03 17:09:50.877362732 +0100
@@ -7,7 +7,7 @@
 	mpeg2t_video.c \
 	mpeg2t_util.c 
 
-INCLUDES = -I$(top_builddir)/include -I$(top_builddir)/lib -I$(top_builddir)/lib/mp4av -I$(top_builddir)/lib/mp4v2 -I$(top_builddir)/lib/SDL/include
+INCLUDES = -I$(top_builddir)/include -I$(top_builddir)/lib -I$(top_builddir)/lib/mp4av -I$(top_builddir)/lib/mp4v2 @SDL_CFLAGS@
 AM_CFLAGS = -DIPTV_COMPATIBLE -D_REENTRANT @BILLS_CWARNINGS@
 AM_CXXFLAGS = -D_REENTRANT @BILLS_CPPWARNINGS@
 
--- mpeg4ip-1.0RC3/lib/Makefile.am.wiget	2002-11-14 21:30:19.000000000 +0100
+++ mpeg4ip-1.0RC3/lib/Makefile.am	2003-12-03 17:09:50.843367929 +0100
@@ -1 +1 @@
-SUBDIRS = avi bitstream config_file fposrec gnu mp4 mp4v2 mp4av msg_queue rtp sdp SDL xvid mpeg2t SDL_VisualC
+SUBDIRS = avi bitstream config_file fposrec gnu mp4 mp4v2 mp4av msg_queue rtp sdp xvid mpeg2t
--- mpeg4ip-1.0RC3/common/video/libmpeg32/video/Makefile.am.wiget	2003-11-13 17:40:15.000000000 +0100
+++ mpeg4ip-1.0RC3/common/video/libmpeg32/video/Makefile.am	2003-12-03 17:09:50.877362732 +0100
@@ -14,7 +14,7 @@
 noinst_LTLIBRARIES = libmpeg3video.la
 libmpeg3video_la_LDFLAGS = -static
 
-INCLUDES=-I$(top_srcdir)/include -I$(top_srcdir)/lib/SDL/include
+INCLUDES=-I$(top_srcdir)/include @SDL_CFLAGS@
 if HAVE_ALIGN_LOOPS
    OUR_ALIGN_LOOPS=@ALIGN_LOOPS@=2
 else
--- mpeg4ip-1.0RC3/common/video/libmpeg32/Makefile.am.wiget	2002-09-24 20:08:00.000000000 +0200
+++ mpeg4ip-1.0RC3/common/video/libmpeg32/Makefile.am	2003-12-03 17:09:50.876362885 +0100
@@ -2,7 +2,7 @@
 
 noinst_LTLIBRARIES = libmpeg3.la libmpeg3util.la
 
-INCLUDES= -I$(top_srcdir)/include -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/SDL/include
+INCLUDES= -I$(top_srcdir)/include -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 @SDL_CFLAGS@
 AM_CFLAGS= -DHAVE_CSS
 libmpeg3util_la_LDFLAGS = -static 
 libmpeg3util_la_SOURCES = \
--- mpeg4ip-1.0RC3/server/mp4live/gui/Makefile.am.wiget	2003-10-13 19:37:54.000000000 +0200
+++ mpeg4ip-1.0RC3/server/mp4live/gui/Makefile.am	2003-12-03 17:09:50.875363037 +0100
@@ -18,7 +18,7 @@
 INCLUDES=@GLIB_CFLAGS@ @GTK_CFLAGS@ \
 	-I.. \
 	-I$(top_srcdir)/include \
-	-I$(top_srcdir)/lib/SDL/include \
+	@SDL_CFLAGS@ \
 	-I$(top_srcdir)/lib/mp4v2 \
 	-I$(top_srcdir)/lib/mp4av \
 	-I$(top_srcdir)/lib/msg_queue \
--- mpeg4ip-1.0RC3/server/mp4live/Makefile.am.wiget	2003-10-10 20:43:13.000000000 +0200
+++ mpeg4ip-1.0RC3/server/mp4live/Makefile.am	2003-12-03 17:09:50.874363190 +0100
@@ -105,7 +105,7 @@
 INCLUDES=\
 	@GLIB_CFLAGS@ @GTK_CFLAGS@ \
 	-I$(top_srcdir)/include \
-	-I$(top_srcdir)/lib/SDL/include \
+	@SDL_CFLAGS@ \
 	-I$(top_srcdir)/lib/mp4v2 \
 	-I$(top_srcdir)/lib/mp4av \
 	-I$(top_srcdir)/lib/msg_queue \
@@ -143,7 +143,6 @@
 	./h261/libmp4live_h261.la \
 	-lpthread \
 	@SDL_LIBS@ \
-	$(top_builddir)/lib/SDL/src/main/libSDLmain.a \
 	$(GUILIBS)
 
 EXTRA_DIST = README.html INTERNALS.html $(ed1) Building_mpeg4ip_with_V4L2_support.html
--- mpeg4ip-1.0RC3/util/yuv/Makefile.am.wiget	2002-07-10 19:00:16.000000000 +0200
+++ mpeg4ip-1.0RC3/util/yuv/Makefile.am	2003-12-03 17:09:50.873363343 +0100
@@ -8,7 +8,7 @@
 
 yuvdisp_SOURCES = yuvdisp.cpp
 
-INCLUDES = -I. -I../../include -I../../lib/SDL/include
+INCLUDES = -I. -I../../include @SDL_CFLAGS@
 
 AM_CFLAGS= -D_REENTRANT @BILLS_CWARNINGS@
 
@@ -16,7 +16,6 @@
 
 LDADD = \
 	@SDL_LIBS@ \
-	$(top_builddir)/lib/SDL/src/main/libSDLmain.a \
-        -lm
+	-lm
 
 EXTRA_DIST=yuvdump60.dsp
--- mpeg4ip-1.0RC3/util/iptv/Makefile.am.wiget	2003-06-27 20:11:50.000000000 +0200
+++ mpeg4ip-1.0RC3/util/iptv/Makefile.am	2003-12-03 17:09:50.873363343 +0100
@@ -2,7 +2,7 @@
 
 iptv_prog_SOURCES = iptv_prog.cpp
 
-INCLUDES = -I. -I../../include -I../../lib/SDL/include \
+INCLUDES = -I. -I../../include @SDL_CFLAGS@ \
 	-I$(top_builddir)/player/lib/libhttp \
 	-I$(top_builddir)/lib/sdp
 
@@ -12,7 +12,6 @@
 
 iptv_prog_LDADD = \
 	@SDL_LIBS@ \
-	$(top_builddir)/lib/SDL/src/main/libSDLmain.a \
 	$(top_builddir)/player/lib/libhttp/libhttp.la \
 	$(top_builddir)/lib/sdp/libsdp.la \
 	$(top_builddir)/lib/gnu/libmpeg4ip_gnu.la \
--- mpeg4ip-1.0RC3/player/lib/rtsp/Makefile.am.wiget	2002-06-17 19:35:12.000000000 +0200
+++ mpeg4ip-1.0RC3/player/lib/rtsp/Makefile.am	2003-12-03 17:09:50.872363496 +0100
@@ -13,7 +13,7 @@
 	rtsp_thread_nx.h \
 	rtsp_util.c 
 
-INCLUDES = -I$(top_builddir)/lib/sdp -I$(top_builddir)/include -I$(top_builddir)/lib/SDL/include -I$(top_builddir)/lib
+INCLUDES = -I$(top_builddir)/lib/sdp -I$(top_builddir)/include @SDL_CFLAGS@ -I$(top_builddir)/lib
 AM_CFLAGS = -DIPTV_COMPATIBLE -D_REENTRANT @BILLS_CWARNINGS@ 
 
 check_PROGRAMS = test test_thread
--- mpeg4ip-1.0RC3/player/lib/audio/mp3/Makefile.am.wiget	2003-11-13 17:40:15.000000000 +0100
+++ mpeg4ip-1.0RC3/player/lib/audio/mp3/Makefile.am	2003-12-03 17:09:50.871363649 +0100
@@ -25,4 +25,4 @@
 	README.LIB 		\
 	libmp360.dsp 
 
-AM_CXXFLAGS = -D_REENTRANT -Werror -Wall -fexceptions -I$(top_srcdir)/lib/SDL/include -I../../../../include
+AM_CXXFLAGS = -D_REENTRANT -Werror -Wall -fexceptions @SDL_CFLAGS@ -I../../../../include
--- mpeg4ip-1.0RC3/player/plugin/audio/raw/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/plugin/audio/raw/Makefile.am	2003-12-03 17:18:41.113175283 +0100
@@ -7,7 +7,7 @@
 raw_audio_plugin_la_LIBADD = \
 	-lm 
 
-INCLUDES = -I$(top_srcdir)/player/src -I$(top_src_dir)/player/lib -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp
+INCLUDES = -I$(top_srcdir)/player/src -I$(top_src_dir)/player/lib @SDL_CFLAGS@ -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/plugin/audio/celp/Makefile.am.wiget	2003-10-22 20:17:34.000000000 +0200
+++ mpeg4ip-1.0RC3/player/plugin/audio/celp/Makefile.am	2003-12-03 17:18:19.627680385 +0100
@@ -10,7 +10,7 @@
 	$(top_srcdir)/player/lib/audio/celp/BS/libcelpbs.la \
 	$(top_srcdir)/player/lib/audio/celp/dec/libcelp.la 
 
-INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/player/lib/audio/celp -I$(top_srcdir)/player/lib/audio/celp/BS -I$(top_srcdir)/player/lib/audio/celp/dec -I$(top_srcdir)/player/lib/audio/celp/include -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/sdp
+INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib @SDL_CFLAGS@ -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/player/lib/audio/celp -I$(top_srcdir)/player/lib/audio/celp/BS -I$(top_srcdir)/player/lib/audio/celp/dec -I$(top_srcdir)/player/lib/audio/celp/include -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/plugin/video/raw/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/plugin/video/raw/Makefile.am	2003-12-03 17:19:55.187097854 +0100
@@ -7,7 +7,7 @@
 raw_video_plugin_la_LIBADD = \
 	-lm 
 
-INCLUDES = -I$(top_srcdir)/player/src -I$(top_src_dir)/player/lib -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp
+INCLUDES = -I$(top_srcdir)/player/src -I$(top_src_dir)/player/lib @SDL_CFLAGS@ -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/plugin/video/xvid/Makefile.am.wiget	2003-11-13 17:40:15.000000000 +0100
+++ mpeg4ip-1.0RC3/player/plugin/video/xvid/Makefile.am	2003-12-03 17:20:00.957157513 +0100
@@ -24,7 +24,7 @@
 	$(top_builddir)/lib/sdp/libsdp.la \
 	$(top_builddir)/lib/mp4av/libmp4av.la
 
-INCLUDES = -I.. -I../.. -I../../../lib -I$(top_srcdir)/lib/SDL/include $(XVID_INC) -I../../../../include -I../../../../lib -I$(top_srcdir)/player/src -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/lib/mp4av
+INCLUDES = -I.. -I../.. -I../../../lib @SDL_CFLAGS@ $(XVID_INC) -I../../../../include -I../../../../lib -I$(top_srcdir)/player/src -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/lib/mp4av
 
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
--- mpeg4ip-1.0RC3/player/plugin/video/mpeg3/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/plugin/video/mpeg3/Makefile.am	2003-12-03 17:19:50.179913922 +0100
@@ -10,7 +10,7 @@
 	$(top_builddir)/common/video/libmpeg32/libmpeg3util.la \
 	@SDL_LIBS@
 
-INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/common/video/libmpeg32 -I$(top_srcdir)/common/video/libmpeg32/video -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp
+INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib @SDL_CFLAGS@ -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/common/video/libmpeg32 -I$(top_srcdir)/common/video/libmpeg32/video -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/plugin/video/h261/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/plugin/video/h261/Makefile.am	2003-12-03 17:19:44.667812335 +0100
@@ -20,7 +20,7 @@
 	-lm \
 	$(top_builddir)/lib/gnu/libmpeg4ip_gnu.la 
 
-INCLUDES = -I.. -I../.. -I../../../lib -I$(top_srcdir)/lib/SDL/include -I../../../../include -I../../../../lib -I$(top_srcdir)/player/src -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/player/plugin/rtp/h261 -I$(top_srcdir)/lib/sdp
+INCLUDES = -I.. -I../.. -I../../../lib @SDL_CFLAGS@ -I../../../../include -I../../../../lib -I$(top_srcdir)/player/src -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/player/plugin/rtp/h261 -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@ -DINT64=uint64_t
 
--- mpeg4ip-1.0RC3/player/plugin/rtp/isma_audio/Makefile.am.wiget	2003-11-13 17:40:15.000000000 +0100
+++ mpeg4ip-1.0RC3/player/plugin/rtp/isma_audio/Makefile.am	2003-12-03 17:09:50.861365177 +0100
@@ -11,7 +11,7 @@
 
 INCLUDES = -I$(top_srcdir)/player/src \
 	   -I$(top_srcdir)/player/lib \
-	   -I$(top_srcdir)/lib/SDL/include \
+	   @SDL_CFLAGS@ \
 	   -I$(top_srcdir)/include \
            -I$(top_srcdir)/lib
 
--- mpeg4ip-1.0RC3/player/plugin/rtp/h261/Makefile.am.wiget	2002-11-07 19:16:03.000000000 +0100
+++ mpeg4ip-1.0RC3/player/plugin/rtp/h261/Makefile.am	2003-12-03 17:09:50.860365330 +0100
@@ -11,7 +11,7 @@
 
 INCLUDES = -I$(top_srcdir)/player/src \
 	   -I$(top_srcdir)/player/lib \
-	   -I$(top_srcdir)/lib/SDL/include \
+	   @SDL_CFLAGS@ \
 	   -I$(top_srcdir)/include \
            -I$(top_srcdir)/lib
 
--- mpeg4ip-1.0RC3/player/src/codec/aa/Makefile.am.wiget	2003-11-13 17:40:15.000000000 +0100
+++ mpeg4ip-1.0RC3/player/src/codec/aa/Makefile.am	2003-12-03 17:13:53.182305122 +0100
@@ -14,7 +14,7 @@
 	-lm
 
 
-INCLUDES =  -I$(top_srcdir)/include -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib/audio -I$(top_srcdir)/player/lib -I$(top_srcdir)/lib/mp4v2
+INCLUDES =  -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib/audio -I$(top_srcdir)/player/lib -I$(top_srcdir)/lib/mp4v2 @SDL_CFLAGS@
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/src/codec/mp3/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/src/codec/mp3/Makefile.am	2003-12-03 17:14:15.721856461 +0100
@@ -21,7 +21,7 @@
 	mp3_rtp_bytestream.cpp \
 	mp3_rtp_bytestream.h
 
-INCLUDES = -I.. -I../..  -I../../../lib -I../../../lib/audio -I$(top_srcdir)/lib/SDL/include -I../../../../common/video/mpeg4 -I../../../../include -I../../../../lib -I$(top_builddir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp
+INCLUDES = -I.. -I../..  -I../../../lib -I../../../lib/audio @SDL_CFLAGS@ -I../../../../common/video/mpeg4 -I../../../../include -I../../../../lib -I$(top_builddir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/src/codec/mpeg4/Makefile.am.wiget	2003-10-23 01:27:23.000000000 +0200
+++ mpeg4ip-1.0RC3/player/src/codec/mpeg4/Makefile.am	2003-12-03 17:14:46.918082875 +0100
@@ -33,7 +33,7 @@
 INCLUDES = -I.. \
 	-I../.. \
 	-I../../../lib \
-	-I$(top_srcdir)/lib/SDL/include \
+	@SDL_CFLAGS@ \
 	-I$(top_builddir)/$(mpeg4_dir) \
 	-I$(top_builddir)/$(mpeg4_dir)/type \
 	-I$(top_builddir)/$(mpeg4_dir)/vtc/include \
--- mpeg4ip-1.0RC3/player/src/codec/wav/Makefile.am.wiget	2003-08-06 00:26:21.000000000 +0200
+++ mpeg4ip-1.0RC3/player/src/codec/wav/Makefile.am	2003-12-03 17:15:13.876957298 +0100
@@ -8,7 +8,7 @@
 
 libdir = @PLAYER_PLUGIN_DIR@
 
-INCLUDES = -I.. -I../..  -I../../../lib -I../../../lib/audio -I$(top_srcdir)/lib/SDL/include -I../../../../common/video/mpeg4 -I../../../../include -I../../../../lib -I$(top_srcdir)/lib/sdp
+INCLUDES = -I.. -I../..  -I../../../lib -I../../../lib/audio @SDL_CFLAGS@ -I../../../../common/video/mpeg4 -I../../../../include -I../../../../lib -I$(top_srcdir)/lib/sdp
 
 AM_CFLAGS = -D_REENTRANT -fexceptions @BILLS_CWARNINGS@
 
--- mpeg4ip-1.0RC3/player/src/Makefile.am.wiget	2003-11-20 22:57:14.000000000 +0100
+++ mpeg4ip-1.0RC3/player/src/Makefile.am	2003-12-03 17:12:44.015886334 +0100
@@ -122,7 +122,7 @@
 	gui_utils.h \
 	gui_xpm.h
 
-INCLUDES = -I. -I$(top_srcdir)/common/video/mpeg4 -I./codec -I./codec/mpeg4 -I../lib -I../lib/audio -I$(top_srcdir)/lib/SDL/include -I$(top_srcdir)/lib/mp4 -I$(top_srcdir)/lib/mp4v2 @ISMACRYPINC@ -I$(top_srcdir)/include -I$(top_srcdir)/include -I$(top_srcdir)/lib/avi -I$(top_srcdir)/lib -I$(top_srcdir)/common/video/libmpeg32 -I$(top_srcdir)/lib/sdp @GLIB_CFLAGS@ @GTK_CFLAGS@ @X_CFLAGS@
+INCLUDES = -I. -I$(top_srcdir)/common/video/mpeg4 -I./codec -I./codec/mpeg4 -I../lib -I../lib/audio -I$(top_srcdir)/lib/mp4 -I$(top_srcdir)/lib/mp4v2 @ISMACRYPINC@ -I$(top_srcdir)/include -I$(top_srcdir)/include -I$(top_srcdir)/lib/avi -I$(top_srcdir)/lib -I$(top_srcdir)/common/video/libmpeg32 -I$(top_srcdir)/lib/sdp @GLIB_CFLAGS@ @GTK_CFLAGS@ @X_CFLAGS@ @SDL_CFLAGS@
 
 AM_CFLAGS= -D_REENTRANT @BILLS_CWARNINGS@ -DPLAYER_PLUGIN_DIR=\"@PLAYER_PLUGIN_DIR@\"
 
@@ -154,8 +154,7 @@
 	libmp4syncsdl.la \
 	libmp4syncbase.la \
 	libmp4playerutils.la \
-	@SDL_LIBS@ \
-	$(top_builddir)/lib/SDL/src/main/libSDLmain.a 
+	@SDL_LIBS@ 
 
 gmp4player_LDADD = $(mp4player_LDADD) \
 	@GTK_LIBS@ @GLIB_LIBS@
--- mpeg4ip-1.0RC3/configure.in.wiget	2003-11-20 23:10:55.000000000 +0100
+++ mpeg4ip-1.0RC3/configure.in	2003-12-03 17:10:54.876577754 +0100
@@ -11,9 +11,14 @@
 AC_CONFIG_SUBDIRS(lib/rtp)
 
 dnl Checks for programs.
-AC_CHECK_FILE(lib/SDL/sdl-config,,
-	      AC_MSG_ERROR([Configure for SDL needs to run first - either run bootstrap or run lib/SDL/configure]))
-
+AC_CHECK_PROG(SDL_CONFIG, sdl-config, sdl-config, none)
+if test "$SDL_CONFIG" = "none"; then
+	AC_MSG_ERROR([sdl-config not found])
+fi
+SDL_CFLAGS=`$SDL_CONFIG --cflags`
+SDL_LIBS=`$SDL_CONFIG --libs`
+AC_SUBST(SDL_CFLAGS)
+AC_SUBST(SDL_LIBS)
 
 AC_LIBTOOL_DLOPEN
 AM_PROG_LIBTOOL
@@ -359,10 +364,6 @@
 fi
 AM_CONDITIONAL(USE_MMX, test x$test_use_mmx = xtrue)
 
-dnl Get SDL cflags and SDL libraries to include
-SDL_LIBS="`sh lib/SDL/sdl-config --libs | sed 's,-L.* -lSDL,\$(top_builddir)/lib/SDL/src/libSDL.la,'`"
-AC_SUBST(SDL_LIBS)
-
 dnl
 BILLS_CWARNINGS="-Wall -Werror -Wmissing-prototypes -Wmissing-declarations -Wno-char-subscripts -Wno-unknown-pragmas"
 BILLS_CPPWARNINGS="-Wall -Werror -Wmissing-prototypes -Wno-char-subscripts -Woverloaded-virtual -Wno-unknown-pragmas -Wno-deprecated"
