--- mpeg4ip-1.0/configure.in.orig	2004-09-09 23:30:59.822404464 +0200
+++ mpeg4ip-1.0/configure.in	2004-09-09 23:31:29.812845224 +0200
@@ -304,14 +304,6 @@
       haveglib20=no
    fi
 fi
-if test x$haveglib20 = xno; then 
-   AM_PATH_GLIB(, [glib_gtk=yes], [glib_gtk=no])
-   if test x$glib_gtk = xyes; then
-      AM_PATH_GTK(, [glib_gtk=yes], [glib_gtk=no])
-      AC_DEFINE(HAVE_GTK, 1, [Have GTK on system])
-   fi
-fi
-
 
 AM_CONDITIONAL(NO_GLIB_GTK, test x$glib_gtk = xno)
 
