diff -urN aa/mpeg4ip-1.0/common/video/mpeg4-2000/idct/idct.hpp mpeg4ip-1.0/common/video/mpeg4-2000/idct/idct.hpp
--- aa/mpeg4ip-1.0/common/video/mpeg4-2000/idct/idct.hpp	2003-10-23 01:26:59.000000000 +0200
+++ mpeg4ip-1.0/common/video/mpeg4-2000/idct/idct.hpp	2004-06-30 18:37:15.963042832 +0200
@@ -22,6 +22,7 @@
 #ifndef _INT_IDCT_
 #define _INT_IDCT_ 1
 
+#include "../type/header.h"
 
 #define IDCT_BLOCK_SIZE	8
 class idct{
--- mpeg4ip-1.1/configure.in.orig	2004-10-22 00:08:39.208353824 +0200
+++ mpeg4ip-1.1/configure.in	2004-10-22 00:10:07.521928120 +0200
@@ -487,7 +487,7 @@
 
 dnl
 BILLS_CWARNINGS="-Wall -Werror -Wmissing-prototypes -Wmissing-declarations -Wno-char-subscripts -Wno-unknown-pragmas"
-BILLS_CPPWARNINGS="-Wall -Werror -Wmissing-prototypes -Wno-char-subscripts -Woverloaded-virtual -Wno-unknown-pragmas -Wno-deprecated"
+BILLS_CPPWARNINGS="-Wall -Werror -Wno-char-subscripts -Woverloaded-virtual -Wno-unknown-pragmas -Wno-deprecated"
 MY_CHECK_CC_OPT([-Wformat=2], mpeg4ip_ok)
 if test $mpeg4ip_ok = yes; then
    BILLS_CWARNINGS="$BILLS_CWARNINGS -Wformat=2"
diff -urN aa/mpeg4ip-1.0/server/mp4live/audio_encoder.h mpeg4ip-1.0/server/mp4live/audio_encoder.h
--- aa/mpeg4ip-1.0/server/mp4live/audio_encoder.h	2003-03-21 02:09:01.000000000 +0100
+++ mpeg4ip-1.0/server/mp4live/audio_encoder.h	2004-06-30 18:46:00.180349688 +0200
@@ -31,17 +31,17 @@
 public:
 	CAudioEncoder() { };
 
-	virtual u_int32_t GetSamplesPerFrame() = NULL;
+	virtual u_int32_t GetSamplesPerFrame() = 0;
 
 	virtual bool EncodeSamples(
 		int16_t* pSamples, 
 		u_int32_t numSamplesPerChannel,
-		u_int8_t numChannels) = NULL;
+		u_int8_t numChannels) = 0;
 
 	virtual bool GetEncodedFrame(
 		u_int8_t** ppBuffer, 
 		u_int32_t* pBufferLength,
-		u_int32_t* pNumSamplesPerChannel) = NULL;
+		u_int32_t* pNumSamplesPerChannel) = 0;
 
 
 	// utility routines
diff -urN aa/mpeg4ip-1.0/server/mp4live/media_codec.h mpeg4ip-1.0/server/mp4live/media_codec.h
--- aa/mpeg4ip-1.0/server/mp4live/media_codec.h	2002-10-30 20:12:37.000000000 +0100
+++ mpeg4ip-1.0/server/mp4live/media_codec.h	2004-06-30 18:45:47.836226280 +0200
@@ -33,11 +33,11 @@
 		m_pConfig = NULL;
 	}
 
-	virtual MediaType GetFrameType(void) = NULL;
+	virtual MediaType GetFrameType(void) = 0;
 
-	virtual bool Init(CLiveConfig* pConfig, bool realTime = true) = NULL;
+	virtual bool Init(CLiveConfig* pConfig, bool realTime = true) = 0;
 
-	virtual void Stop() = NULL;
+	virtual void Stop() = 0;
 
 protected:
 	CLiveConfig*	m_pConfig;
diff -urN aa/mpeg4ip-1.0/server/mp4live/media_flow.h mpeg4ip-1.0/server/mp4live/media_flow.h
--- aa/mpeg4ip-1.0/server/mp4live/media_flow.h	2003-10-14 18:02:15.000000000 +0200
+++ mpeg4ip-1.0/server/mp4live/media_flow.h	2004-06-30 18:43:35.501344240 +0200
@@ -37,9 +37,9 @@
 		m_started = false;
 	}
 
-	virtual void Start() = NULL;
+	virtual void Start() = 0;
 
-	virtual void Stop() = NULL; 
+	virtual void Stop() = 0; 
 
 	void SetConfig(CLiveConfig* pConfig) {
 		m_pConfig = pConfig;
diff -urN aa/mpeg4ip-1.0/server/mp4live/media_node.h mpeg4ip-1.0/server/mp4live/media_node.h
--- aa/mpeg4ip-1.0/server/mp4live/media_node.h	2003-12-04 01:53:58.000000000 +0100
+++ mpeg4ip-1.0/server/mp4live/media_node.h	2004-06-30 18:43:07.006676088 +0200
@@ -84,7 +84,7 @@
 	static const uint32_t MSG_NODE_STOP		= MSG_NODE + 2;
 	static const uint32_t MSG_NODE_STOP_THREAD	= MSG_NODE + 3;
 
-	virtual int ThreadMain(void) = NULL;
+	virtual int ThreadMain(void) = 0;
 
 protected:
 	SDL_Thread*			m_myThread;
diff -urN aa/mpeg4ip-1.0/server/mp4live/media_source.h mpeg4ip-1.0/server/mp4live/media_source.h
--- aa/mpeg4ip-1.0/server/mp4live/media_source.h	2003-12-04 01:53:58.000000000 +0100
+++ mpeg4ip-1.0/server/mp4live/media_source.h	2004-06-30 18:43:19.674750248 +0200
@@ -64,11 +64,11 @@
 			NULL, 0, m_myMsgQueueSemaphore);
 	}
 
-	virtual bool IsDone() = NULL;
+	virtual bool IsDone() = 0;
 
 	virtual Duration GetElapsedDuration();
 
-	virtual float GetProgress() = NULL;
+	virtual float GetProgress() = 0;
 
 	virtual u_int32_t GetNumEncodedVideoFrames() {
 		return m_videoEncodedFrames;
--- mpeg4ip-1.1/server/mp4live/video_encoder.h.orig	2004-04-08 20:07:39.000000000 +0200
+++ mpeg4ip-1.1/server/mp4live/video_encoder.h	2004-10-22 00:10:49.934480432 +0200
@@ -31,21 +31,21 @@
 public:
 	CVideoEncoder() { };
 
-	virtual bool Init(CLiveConfig *pConfig, bool realTime = true) = NULL;
+	virtual bool Init(CLiveConfig *pConfig, bool realTime = true) = 0;
 
 	virtual bool EncodeImage(
 		u_int8_t* pY, u_int8_t* pU, u_int8_t* pV,
 		u_int32_t yStride, u_int32_t uvStride,
 		bool wantKeyFrame,
 		Duration elapsedDuration,
-		Timestamp srcFrameTimestamp) = NULL;
+		Timestamp srcFrameTimestamp) = 0;
 
 	virtual bool GetEncodedImage(
 		u_int8_t** ppBuffer, u_int32_t* pBufferLength,
-		Timestamp *dts, Timestamp *pts) = NULL;
+		Timestamp *dts, Timestamp *pts) = 0;
 
 	virtual bool GetReconstructedImage(
-		u_int8_t* pY, u_int8_t* pU, u_int8_t* pV) = NULL;
+		u_int8_t* pY, u_int8_t* pU, u_int8_t* pV) = 0;
 	virtual media_free_f GetMediaFreeFunction(void) { return NULL; };
 
 	virtual bool CanGetEsConfig (void) { return false; };
