--- mpeg4ip-1.6.1/player/plugin/audio/ffmpeg/ffmpeg.h~	2007-07-02 22:26:01.000000000 +0200
+++ mpeg4ip-1.6.1/player/plugin/audio/ffmpeg/ffmpeg.h	2008-08-26 17:07:24.000000000 +0200
@@ -26,7 +26,7 @@
 #include "codec_plugin.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
--- mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.h~	2007-07-02 22:26:04.000000000 +0200
+++ mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.h	2008-08-26 17:15:49.000000000 +0200
@@ -25,7 +25,7 @@
 #include "audio_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
--- mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.h~	2007-07-02 22:26:05.000000000 +0200
+++ mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.h	2008-08-26 17:16:07.000000000 +0200
@@ -25,7 +25,7 @@
 #include "video_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
--- mpeg4ip-1.6.1/configure.in.orig	2008-08-26 17:02:49.000000000 +0200
+++ mpeg4ip-1.6.1/configure.in	2008-08-26 19:28:09.000000000 +0200
@@ -487,7 +487,7 @@
   if test x$have_ffmpeg_arg = xtrue; then
      AC_MSG_NOTICE([have enable ffmpeg $enable_ffmpeg])
      FFMPEG_INC=-I${enable_ffmpeg}/libavcodec
-     FFMPEG_LIB="${enable_ffmpeg}/libavcodec/libavcodec.a -lz"
+     FFMPEG_LIB="-lavcodec -lavutil -lz"
      saveCFLAGS="$CFLAGS"
      CFLAGS="$CFLAGS $FFMPEG_INC"
      AC_CHECK_TYPES([AVRational], , , [#include <avcodec.h>])
diff -burNp mpeg4ip-1.6.1/configure.in mpeg4ip-1.6.1-new/configure.in
--- mpeg4ip-1.6.1/configure.in	2007-09-28 22:45:08.000000000 +0200
+++ mpeg4ip-1.6.1-new/configure.in	2008-10-10 22:09:56.469368599 +0200
@@ -512,7 +512,7 @@ if test x$disable_ffmpeg = xfalse; then
      if test x$have_ffmpeg = xtrue; then
         AC_DEFINE(HAVE_FFMPEG_INSTALLED, 1, [defined if ffmpeg is installed])
         FFMPEG_INC=
-        FFMPEG_LIB="-lavcodec -lz $ffmpeg_add_libs"
+        FFMPEG_LIB="-lavcodec -lavutil -lz $ffmpeg_add_libs $(pkg-config libswscale --libs --silence-errors)"
         AC_CHECK_TYPES([AVRational], , , [#include <ffmpeg/avcodec.h>])
 	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <ffmpeg/avcodec.h>])
 	AC_CHECK_DECLS([avcodec_decode_audio2], , ,[#include <ffmpeg/avcodec.h>])
--- mpeg4ip-1.6.1/player/plugin/video/ffmpeg/ffmpeg.h	2007-07-02 22:26:03.000000000 +0200
+++ mpeg4ip-1.6.1-new/player/plugin/video/ffmpeg/ffmpeg.h	2008-10-10 22:16:05.176622742 +0200
@@ -27,7 +27,8 @@
 #include "codec_plugin.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 #else
 #include <avcodec.h>
 #endif
--- mpeg4ip-1.6.1/player/plugin/video/ffmpeg/ffmpeg.cpp	2007-07-02 22:26:03.000000000 +0200
+++ mpeg4ip-1.6.1-new/player/plugin/video/ffmpeg/ffmpeg.cpp	2008-10-10 23:01:19.519476374 +0200
@@ -186,7 +186,6 @@
 
   ffmpeg->m_vft = vft;
   ffmpeg->m_ifptr = ifptr;
-  avcodec_init();
   avcodec_register_all();
   av_log_set_level(AV_LOG_QUIET);
 
@@ -585,10 +585,16 @@ static int ffmpeg_decode (codec_data_t *
 	from.data[ix] = ffmpeg->m_picture->data[ix];
 	from.linesize[ix] = ffmpeg->m_picture->linesize[ix];
       }
-      
-      img_convert(&to, PIX_FMT_YUV420P,
-		  &from, ffmpeg->m_c->pix_fmt,
-		  ffmpeg->m_c->width, ffmpeg->m_c->height);
+#ifdef HAVE_FFMPEG_INSTALLED
+      SwsContext *pSWSCtx;
+      pSWSCtx = sws_getContext(ffmpeg->m_c->width, ffmpeg->m_c->height,
+			       ffmpeg->m_c->pix_fmt,
+			       ffmpeg->m_c->width, ffmpeg->m_c->height,
+			       PIX_FMT_YUV420P, SWS_BICUBIC, 0, 0, 0);
+      sws_scale(pSWSCtx, from.data, from.linesize, 0, ffmpeg->m_c->height,
+		to.data, to.linesize);
+      sws_freeContext(pSWSCtx);
+#endif
       ffmpeg->m_vft->video_filled_buffer(ffmpeg->m_ifptr,
 					 ffmpeg->have_cached_ts ?
 					 ffmpeg->cached_ts : ts);
@@ -635,7 +634,6 @@
 {
   enum CodecID fcodec;
   AVCodec *c;
-  avcodec_init();
   avcodec_register_all();
   av_log_set_level(AV_LOG_QUIET);
   fcodec = ffmpeg_find_codec(stream_type, compressor, type, profile, 
--- mpeg4ip-1.6.1/player/plugin/audio/ffmpeg/ffmpeg.cpp~	2011-07-26 18:37:06.785253535 +0200
+++ mpeg4ip-1.6.1/player/plugin/audio/ffmpeg/ffmpeg.cpp	2011-07-26 18:43:29.810426398 +0200
@@ -135,7 +135,6 @@
 
   ffmpeg->m_vft = vft;
   ffmpeg->m_ifptr = ifptr;
-  avcodec_init();
   avcodec_register_all();
 
   ffmpeg->m_codecId = ffmpeg_find_codec(stream_type, compressor, type, 
@@ -220,14 +220,13 @@
   uint32_t freq_ts = pts->audio_freq_timestamp;
 
   do {
-#if HAVE_DECL_AVCODEC_DECODE_AUDIO2 != 1
-    used = avcodec_decode_audio(ffmpeg->m_c, (short *)ffmpeg->m_outbuf,
-				&outsize, buffer, left);
-#else
-    outsize = AVCODEC_MAX_AUDIO_FRAME_SIZE;
-    used = avcodec_decode_audio2(ffmpeg->m_c, (int16_t *)ffmpeg->m_outbuf,
-				 &outsize, buffer, left);
-#endif
+	AVPacket pkt;
+	av_init_packet(&pkt);
+	pkt.data = buffer; 
+	pkt.size = left; 
+	used = avcodec_decode_audio3(ffmpeg->m_c, (int16_t *)ffmpeg->m_outbuf,
+			&outsize, &pkt);
+
     if (used < 0) {
       ffmpeg_message(LOG_DEBUG, "ffmpeg", "failed to decode at "U64, 
 		     ts);
@@ -288,7 +288,6 @@
 {
   enum CodecID fcodec;
   AVCodec *c;
-  avcodec_init();
   avcodec_register_all();
 
   fcodec = ffmpeg_find_codec(stream_type, compressor, type, profile, 
--- mpeg4ip-1.6.1/player/plugin/video/ffmpeg/ffmpeg.cpp.orig	2011-07-26 13:26:04.684844016 +0200
+++ mpeg4ip-1.6.1/player/plugin/video/ffmpeg/ffmpeg.cpp	2011-07-26 21:00:12.921661697 +0200
@@ -295,11 +295,13 @@
       uint32_t offset = 0;
       do {
 	int got_picture;
-	offset += avcodec_decode_video(ffmpeg->m_c, 
-				       ffmpeg->m_picture,
-				       &got_picture,
-				       (uint8_t *)userdata + offset, 
-				       ud_size - offset);
+	AVPacket pkt;
+	av_init_packet(&pkt);
+	pkt.data = (uint8_t *)userdata + offset;
+	pkt.size = ud_size - offset;
+	offset += avcodec_decode_video2(ffmpeg->m_c,
+			ffmpeg->m_picture,
+			&got_picture, &pkt);
       } while (offset < ud_size);
     }
 	
@@ -442,11 +444,13 @@
   int ret;
   do {
     int local_got_picture;
-    ret = avcodec_decode_video(ffmpeg->m_c, 
-			       ffmpeg->m_picture,
-			       &local_got_picture,
-			       buffer + bytes_used, 
-			       buflen - bytes_used);
+	AVPacket pkt;
+	av_init_packet(&pkt);
+	pkt.data = buffer + bytes_used; 
+	pkt.size = buflen - bytes_used;
+	ret = avcodec_decode_video2(ffmpeg->m_c,
+			ffmpeg->m_picture,
+			&local_got_picture, &pkt);
     bytes_used += ret;
     //ffmpeg_message(LOG_CRIT, "ffmpeg", "used %d %d", ret, local_got_picture);
     got_picture |= local_got_picture;
--- mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.cpp~	2006-09-29 21:55:42.000000000 +0200
+++ mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.cpp	2011-07-26 22:13:07.825014397 +0200
@@ -442,7 +442,6 @@
 {
   const char *encoding = Profile()->GetStringValue(CFG_AUDIO_ENCODING);
 
-  avcodec_init();
   avcodec_register_all();
 
   if (strcasecmp(encoding,AUDIO_ENCODING_MP3) == 0) {
@@ -475,7 +475,7 @@
   m_avctx = avcodec_alloc_context();
   m_frame = avcodec_alloc_frame();
 
-  m_avctx->codec_type = CODEC_TYPE_AUDIO;
+  m_avctx->codec_type = AVMEDIA_TYPE_AUDIO;
   switch (m_media_frame) {
   case MP3AUDIOFRAME:
     m_avctx->codec_id = CODEC_ID_MP2;
@@ -647,7 +646,6 @@
   AddAudioEncoderTable(&ffmpeg_alaw_audio_encoder_table);
   AddAudioEncoderTable(&ffmpeg_ulaw_audio_encoder_table);
 #ifdef MAY_HAVE_AMR_CODEC
-  avcodec_init();
   avcodec_register_all();
   bool have_amr_nb = avcodec_find_encoder(CODEC_ID_AMR_NB) != NULL;
   bool have_amr_wb = avcodec_find_encoder(CODEC_ID_AMR_WB) != NULL;
--- mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.cpp~	2007-01-30 22:53:45.000000000 +0100
+++ mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.cpp	2011-07-26 22:17:55.917224166 +0200
@@ -72,7 +72,6 @@
 
 bool CFfmpegVideoEncoder::Init (void)
 {
-  avcodec_init();
   avcodec_register_all();
 
   if (m_push != NULL) {
@@ -222,9 +222,9 @@
       m_count = 0;
     }
   }
-  if (wantKeyFrame) m_picture->pict_type = FF_I_TYPE; //m_picture->key_frame = 1;
+  if (wantKeyFrame) m_picture->pict_type = AV_PICTURE_TYPE_I; //m_picture->key_frame = 1;
   else //m_picture->key_frame = 0;
-    m_picture->pict_type = 0;
+    m_picture->pict_type = AV_PICTURE_TYPE_NONE;
 
   m_picture->data[0] = (uint8_t *)pY;
   m_picture->data[1] = (uint8_t *)pU;
