--- mpeg4ip-1.0/common/video/libmpeg32/video/Makefile.am.orig	2004-02-11 10:00:20.000000000 +0000
+++ mpeg4ip-1.0/common/video/libmpeg32/video/Makefile.am	2004-02-11 11:00:07.000000000 +0000
@@ -12,7 +12,6 @@
 ###########################################################################
 
 noinst_LTLIBRARIES = libmpeg3video.la
-libmpeg3video_la_LDFLAGS = -static
 
 INCLUDES=-I$(top_srcdir)/include @SDL_CFLAGS@
 if HAVE_ALIGN_LOOPS
--- mpeg4ip-1.0/player/lib/audio/celp/dec/Makefile.am.orig	2002-06-05 19:15:05.000000000 +0000
+++ mpeg4ip-1.0/player/lib/audio/celp/dec/Makefile.am	2004-02-11 10:57:57.000000000 +0000
@@ -1,8 +1,6 @@
 
 noinst_LTLIBRARIES = libcelp.la
 
-libcelp_la_LDFLAGS = -static
-
 libcelp_la_SOURCES = \
 	att_abs_postp.c \
 	att_bwx.c \
--- mpeg4ip-1.0/player/lib/audio/faad/Makefile.am.orig	2003-10-03 23:55:12.000000000 +0000
+++ mpeg4ip-1.0/player/lib/audio/faad/Makefile.am	2004-02-11 11:02:48.000000000 +0000
@@ -1,5 +1,4 @@
 noinst_LTLIBRARIES = libfaad.la
-libfaad_la_LDFLAGS = -static
 libfaad_la_SOURCES = \
 	bits.c \
 	block.c \
